pkg/declarativeconfig/configuration.go

```diff
@@ -111,3 +100,37 @@ func decodeYAMLToConfiguration(rawYAML []byte, configuration Configuration) erro
 	}
 	return nil
 }
+
+func parseToConfiguration(contents []byte) ([]Configuration, error) {
+	dec := yaml.NewDecoder(bytes.NewReader(contents))
+	var unstructuredObjs []interface{}
+	for {
```

Comment:
WDYT about putting some iteration limit on this `for`, like 10000 iterations?

Reply:
What's the reasoning behind this?
In case there's an issue with decoding, we will exit early.

I don't think we can come up with a reasonable number here, since this is correlating with the YAML file you are parsing configuration from, and there usually isn't a hard limit on that.